<launch>
    <!-- =========================================================================================================== -->
    <!-- ===========================================NODE SETTINGS=================================================== -->
    <!-- =========================================================================================================== -->
    <!-- Global Settings For All Nodes -->
    <arg name="2D_camera_name" default="linear_3dof_2d_camera"/>
    <arg name="3D_camera_name" default="linear_3dof_3d_camera"/>
    <arg name="gripper_name" default="linear_3dof_gripper"/>
    <arg name="odom_frame_id" default="linear_arm_3dof_home"/>

    <!-- =========================================================================================================== -->
    <!-- ===============================================NODES======================================================= -->
    <!-- =========================================================================================================== -->
    <!-- Launch robot state publisher -->
    <node pkg="tf" type="static_transform_publisher" name="base_link_publisher" args="0 0 0 0 0 0 map base_link 100" />
    <node pkg="tf" type="static_transform_publisher" name="linear_arm_3dof_home_tf_publisher" args="0 0 0 0 0 0 map
    $(arg odom_frame_id) 100" />

    <param name="linear_arm_3dof_robot_description" command="$(find xacro)/xacro --inorder '$(find grasberry_manipulation)/description/linear_arm_3dof.urdf.xacro'"/>
    <node name="linear_arm_3dof_robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
        <remap from="robot_description" to="linear_arm_3dof_robot_description" />
    </node>

    <!-- Launch Arm Controller -->
    <node pkg="rasberry_perception" name="rasberry_perception_arm_controller" type="arm_controller.py" output="screen" required="true">
        <param name="2D_camera_frame" value="$(arg 2D_camera_name)_link"/>
        <param name="3D_camera_frame" value="$(arg 3D_camera_name)_link"/>
        <param name="gripper_frame" value="$(arg gripper_name)_link"/>
        <param name="odom_frame_id" value="$(arg odom_frame_id)"/>
    </node>

    <!-- Launch Gripper Controller -->
    <!-- Open ROS Serial interface -->
    <node pkg="rosserial_python" type="serial_node.py" name="serial_node_gripper" output="screen">
        <param name="port" value="/dev/ttyUSB0"/>
        <param name="baud" value="57600"/>
    </node>

    <node pkg="rosserial_python" type="serial_node.py" name="serial_node_ir" output="screen">
        <param name="port" value="/dev/ttyUSB1"/>
        <param name="baud" value="57600"/>
    </node>

    <node pkg="rasberry_perception" name="rasberry_perception_gripper_service" type="gripper_controller.py" output="screen"/>
</launch>
